source/tests/pt/model/test_dpa1.py

```diff
@@ -0,0 +1,214 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+import itertools
+import unittest
+
+import numpy as np
+import torch
+
+try:
+    from deepmd.dpmodel.descriptor.dpa1 import DescrptDPA1 as DPDescrptDPA1
+
+    support_se_atten = True
+except ModuleNotFoundError:
+    support_se_atten = False
+except ImportError:
+    support_se_atten = False
+
+from deepmd.pt.model.descriptor.dpa1 import (
+    DescrptDPA1,
+)
+from deepmd.pt.utils import (
+    env,
+)
+from deepmd.pt.utils.env import (
+    PRECISION_DICT,
+)
+
+from .test_env_mat import (
+    TestCaseSingleFrameWithNlist,
+)
+from .test_mlp import (
+    get_tols,
+)
+
+dtype = env.GLOBAL_PT_FLOAT_PRECISION
+
+
+@unittest.skipIf(not support_se_atten, "EnvMat not supported")
+class TestDescrptSeAtten(unittest.TestCase, TestCaseSingleFrameWithNlist):
+    def setUp(self):
+        TestCaseSingleFrameWithNlist.setUp(self)
+
+    def test_consistency(
+        self,
+    ):
+        rng = np.random.default_rng(100)
+        nf, nloc, nnei = self.nlist.shape
+        davg = rng.normal(size=(self.nt, nnei, 4))
+        dstd = rng.normal(size=(self.nt, nnei, 4))
+        dstd = 0.1 + np.abs(dstd)
+
+        for idt, prec, sm, to in itertools.product(
```

Comment:
add comment explaining the meaning of the variables.

Reply:
I find the test may not fully cover all the possibilities of the input parameters of dpa1. 
1. type_one_side
2. attn_layer 0 or > 0
3. attn_dotr
4. attn_mask (do you really use it??) 
5. exclude types
6. set_davg_zero
7. scaling_factor
8. normalize
9. temperature (really used?)
10. smooth_type_embedding
11. concat_output_tebd

Reply:
1，2，5，6，10，11 are used to be tested in source/tests/consistent/descriptor/test_dpa1.py

Reply:
All params above, except `attn_mask`, are added to consistent tests.